app/views/unattended/provisioning_templates/snippet/built.erb

```
@@ -39,13 +39,18 @@ EOF
 -%>
 <% end -%>
 if [ -x /usr/bin/curl ]; then
-  /usr/bin/curl -o /dev/null <%= curl_opts.join(' ') %> --silent '<%= url %>'
+  /usr/bin/curl -o /dev/null <%= curl_opts.join(' ') %> --silent --show-error '<%= url %>'
 elif [ -x /usr/bin/wget ]; then
   /usr/bin/wget -q -O /dev/null --method <%= method %> <%= wget_opts.join(' ') %> '<%= url %>'
 else
   wget -q -O /dev/null --header 'Content-Type: text/plain' '<%= url %>'
 fi
 
-if [ -x "$(command -v subscription-manager)" ] ; then
+if [[ $? == 0  && -x "$(command -v subscription-manager)" ]] ; then
+  echo "Host [<%= @host.name %>] successfully configured."
   subscription-manager facts --update
+else
+  echo "Failed to set built status for host [<%= @host.name %>]."
+  exit 1
 fi
```

This would effectively make subscription-manager a hard dependency, wouldn't it? That might be a breaking change for non-katello deployments.

---

If i am not wrong, that the condition about "command -v sub-man" has always been present ( since last year ).

Can you explain a bit about your comment in case I misunderstood your concern here ?

---

Previously it was "if subscription manager is available, do something extra otherwise do nothing", now it is "if previous command succeeded and subman is available, do something extra, otherwise fail hard", right? So if the previous command succeeds, but subman is not available, it will fail.

> Can you explain a bit about your comment in case I misunderstood your concern here ?

If you have a vanilla foreman without katello, you can register systems which don't have subscription manager at all. This change would make that impossible

---

Ah okay.. Yes and Correct.. So I am assuming this `fail hard` part is what you are most concerned about i.e. the `exit 1`? 

I can remove it and it would not be affecting the actual goal too much. 

Otherwise, let me know if you have any specific suggestions here

---

I've pushed a small improvement. Let me know what you think about that. 

With http blocked, the build error is reflected, and the last sub-man repos command is also executed

~~~
..
..
Refreshing subscription data
All local data refreshed
curl: (7) Failed to connect to foremaninstance-fqdn port 80: No route to host
Failed to set built status for host [client-host-fqdn].
+----------------------------------------------------------+
    Available Repositories in /etc/yum.repos.d/redhat.repo
+----------------------------------------------------------+
..
..
~~~

With http open, build success is reflected and the last sub-man repos command is also executed along with facts update. 

~~~
..
..
Refreshing subscription data
All local data refreshed
Host [client-host-fqdn] successfully configured.
Successfully updated the system facts.
+----------------------------------------------------------+
    Available Repositories in /etc/yum.repos.d/redhat.repo
+----------------------------------------------------------+
..
..
~~~

---

@adamruzicka ^^

---

Hi @adamruzicka, it seems we have [#10555](https://github.com/theforeman/foreman/pull/10555) as well, which was raised much earlier than mine and has a bit different approach as well.